### PR TITLE
Use the project CRS on basemap generation, not the layer CRS

### PR DESCRIPTION
**Pull request by suricactus**

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-1445 QFieldSync fix basemap generation extent](https://app.clickup.com/t/2192114/QF-1445)

File: offline_converter.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def __init__(
         self.area_of_interest.fromWkt(area_of_interest_wkt)
         self.area_of_interest_crs = QgsCoordinateReferenceSystem(area_of_interest_epsg)
 
-        assert self.area_of_interest.isValid()
+        assert self.area_of_interest.isValid()[0]
```
[the remark this answers is not in the document]
Comment (suricactus) [reply offline_converter.py:114]: tuple[bool, str]

[the remark this answers is not in the document]
Comment (suricactus) [reply offline_converter.py:114]: https://qgis.org/api/classQgsSurface.html#ac5a40d1eec03998eb3e07b4494b3ab51 when a srting is passed by reference and modified by the call, it's common practice to return a tuple, especially in pyqgis

[the remark this answers is not in the document]
Comment (m-kuhn) [reply offline_converter.py:114]: Ah, ok.

What I often like to do in such cases to keep it readable:

```
is_valid, _ = self.area_of_interest.isValid()
assert is_valid
```

Feel free to keep the short or the verbose version and merge ahead at your descretion

**Review by m-kuhn — APPROVED** [review]

File: offline_converter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (suricactus) [reply offline_converter.py:114]: yes, I do agree, but in this case I use it in the assert only and will be a bit too verbose

**Pull request merged**